addons/mrp/mrp.py

```diff
@@ -155,6 +155,12 @@ class mrp_bom(osv.osv):
     _description = 'Bill of Material'
     _inherit = ['mail.thread']
 
+    def _get_bom_ids(self, cr, uid, ids, context=None):
+        result = {}
+        for bom in self.pool.get('product.template').browse(cr, uid, ids, context=context).bom_ids:
+            result[bom.id] = True
+        return result.keys()
```

Comment:
Why building a dictionary for returning only its keys? Why not a list of bom.id's?